lib/nmatrix/math.rb

```diff
@@ -881,6 +880,41 @@ def nrm2 incx=1, n=nil
   end
   alias :norm2 :nrm2
 
+  #
+  # call-seq:
+  #     scale! -> NMatrix
+  #
+  # == Arguments
+  #   - +alpha+ -> Scalar value used in the operation.
+  #   - +inc+ -> Increment used in the scaling function. Should generally be 1.
+  #   - +n+ -> Number of elements of +vector+.
+  #
+  # This is a destructive method, modifying the source NMatrix.  See also #scale.
+  # Return the scaling result of the matrix. BLAS scal will be invoked if provided.
+
+  def scale!(alpha, incx=1, n=nil)
```

Comment:
Actually, I took nrm2 as the reference, which has such two arguments. I think incx and n both are internal parameters so they may not be exposed as the arguments in the outer method from the user's viewpoint.